### PR TITLE
Extract: flags, delay, ds-help

### PR DESCRIPTION
**Pull request by hsanjuan**
Extract: `flags` and `thirdparty/delay` and `thirdparty/ds-help` submodules

They have been moved to their own repositories:
    
* https://github.com/ipfs/go-ipfs-delay
* https://github.com/ipfs/go-ipfs-flags
* https://github.com/ipfs/go-ipfs-ds-help
    
History has been preserved. They have been published
with gx. Imports have been updated and re-ordered
accordingly.
    
License: MIT
Signed-off-by: Hector Sanjuan <hector@protocol.ai>

### CODE REVIEW

**Comment by magik6k** [comment]
You could try to pass the low-mem flag to bitswap constructor to avoid having to create a module with 13 lines of code, which seems to be a bit of an overkill.

**Comment by hsanjuan** [comment]
It is my understanding that this module was created so other modules could re-use common flags. It might be overkill because there is only one flag used by one other module, but I can only guess this was done this way for a reason, foreseeing that this could grow or at least set a precedent on how to do such things.

Any ways, I am not re-factoring code, I'm just extracting.

**Comment by whyrusleeping** [comment]
Yeah, The idea is that we would have many packages reusing those flags. The initiative hasnt yet taken off, but I think we should loop back into that at some point.

**Comment by whyrusleeping** [comment]
@hsanjuan gonna need a rebase

**Comment by hsanjuan** [comment]
@whyrusleeping rebased.

File: exchange/bitswap/bitswap.go [hunk 1]
```diff
@@ -16,9 +16,9 @@ import (
 	bsmsg "github.com/ipfs/go-ipfs/exchange/bitswap/message"
 	bsnet "github.com/ipfs/go-ipfs/exchange/bitswap/network"
 	notifications "github.com/ipfs/go-ipfs/exchange/bitswap/notifications"
-	flags "github.com/ipfs/go-ipfs/flags"
-	"github.com/ipfs/go-ipfs/thirdparty/delay"
 
+	"gx/ipfs/QmRJVNatYJwTAHgdSM1Xef9QVQ1Ch3XHdmcrykjP5Y4soL/go-ipfs-delay"
```
Comment (magik6k) [inline exchange/bitswap/bitswap.go]: I'd prefix this too for consistency

File: exchange/bitswap/testutils.go [hunk 2]
```diff
@@ -7,7 +7,7 @@ import (
 	blockstore "github.com/ipfs/go-ipfs/blocks/blockstore"
 	tn "github.com/ipfs/go-ipfs/exchange/bitswap/testnet"
 	datastore2 "github.com/ipfs/go-ipfs/thirdparty/datastore2"
-	delay "github.com/ipfs/go-ipfs/thirdparty/delay"
+	delay "gx/ipfs/QmRJVNatYJwTAHgdSM1Xef9QVQ1Ch3XHdmcrykjP5Y4soL/go-ipfs-delay"
```
Comment (magik6k) [inline exchange/bitswap/testutils.go]: Move to the gx/ group

File: filestore/fsrefstore.go [hunk 3]
```diff
@@ -9,15 +9,15 @@ import (
...
-	"gx/ipfs/Qmej7nf81hi2x2tvjRBF3mcp74sQyuDH4VMYDGd1YtXjb2/go-block-format"
 
 	ds "gx/ipfs/QmPpegoMqhAEqjncrzArm7KVWAkCm78rqL2DPuNjhPrshg/go-datastore"
 	dsns "gx/ipfs/QmPpegoMqhAEqjncrzArm7KVWAkCm78rqL2DPuNjhPrshg/go-datastore/namespace"
 	dsq "gx/ipfs/QmPpegoMqhAEqjncrzArm7KVWAkCm78rqL2DPuNjhPrshg/go-datastore/query"
 	proto "gx/ipfs/QmT6n4mspWYEya864BhCUJEgyxiRfmiSY9ruQwTUNpRKaM/protobuf/proto"
+	posinfo "gx/ipfs/Qmb3jLEFAQrqdVgWUajqEyuuDoavkSq1XQXz6tWdFWF995/go-ipfs-posinfo"
 	cid "gx/ipfs/QmcZfnkapfECQGcLZaf9B79NRg7cRa9EnZh4LSbkCzwNvY/go-cid"
+	dshelp "gx/ipfs/QmdQTPWduSeyveSxeCAte33M592isSW5Z979g81aJphrgn/go-ipfs-ds-help"
+	"gx/ipfs/Qmej7nf81hi2x2tvjRBF3mcp74sQyuDH4VMYDGd1YtXjb2/go-block-format"
```
Comment (magik6k) [inline filestore/fsrefstore.go]: prefix

**Comment by hsanjuan** [comment]
Thanks @magik6k !

**Pull request merged**